modules/sharedaddy/admin-sharing.js

```diff
@@ -123,9 +123,12 @@
 				$( '#live-preview div.sharedaddy' ).addClass( 'sd-social-icon' );
 			} else if ( 'official' === button_style ) {
 				$( '#live-preview ul.preview .advanced, .sharing-hidden .inner ul .advanced' ).each( function( /*i*/ ) {
-					if ( !$( this ).hasClass( 'preview-press-this' ) && !$( this ).hasClass( 'preview-email' ) && !$( this ).hasClass( 'preview-print' ) && !$( this ).hasClass( 'preview-telegram' ) && !$( this ).hasClass( 'share-custom' ) ) {
+					if ( !$( this ).hasClass( 'preview-press-this' ) && !$( this ).hasClass( 'preview-email' )
+              && !$( this ).hasClass( 'preview-print' ) && !$( this ).hasClass( 'preview-telegram-me' ) && !$( this ).hasClass( 'share-custom' ) ) {
 						$( this ).find( '.option a span' ).html( '' ).parent().removeClass( 'sd-button' ).parent().attr( 'class', 'option option-smart-on' );
-					}
+					} else if ( $( this ).hasClass( 'preview-telegram-me' ) ) {
```

Comment:
This is the JS that drives the "smart" preview of the official buttons (I have a screenshot of the official buttons above).

Telegram doesn't have an official button, so I built one with CSS to match the other official buttons. This makes it unique, all the other buttons either:
- Use hosted button code that we can't preview. A static image is used instead. This is the general case the JS code is written for.
- Don't change, and remain monochrome when in official button mode, eg: email, print. These require a special case in the JS.

Rather than adding a static image of something I'm already creating with CSS, I opted to use JS to add the appropriate classes so the CSS can display properly.